src/raft.c

```diff
@@ -1640,18 +1741,49 @@ void RaftReqFree(RaftReq *req)
         RedisModule_FreeThreadSafeContext(req->ctx);
         RedisModule_UnblockClient(req->client, NULL);
     }
+
     RedisModule_Free(req);
 }
 
-RaftReq *RaftReqInit(RedisModuleCtx *ctx, enum RaftReqType type)
+void blockedTimedOut(RedisModuleCtx *ctx, void *data)
+{
+    RedisRaftCtx *rr = &redis_raft;
+    RaftReq *req = data;
+
+    /* don't need to attach the req to this entry, as the req is part of the BlockedClient object */
+    raft_entry_t *entry = raft_entry_new(sizeof(raft_index_t));
+    entry->id = rand();
+    entry->type = RAFT_LOGTYPE_TIMEOUT_BLOCKED;
+    memcpy(entry->data, &req->raft_idx, entry->data_len);
```

Comment:
One thing that slightly bothers me here is we use binary serialization and it won't be readable on file. 
We use string serialization normally (only exception is RaftCfgChange unfortunately). 

Here it looks easier to switch to string serialization as it is just a number (probably you can use `decodeInteger()/encodeInteger()`. Maybe we should stay consistent and use string serialization here and we can do it for RaftCfgChange later?

Reply:
ok

Reply:
I think we can fix multibulk format later. For now, let's just try to serialize things as strings.